### PR TITLE
[TEVA-1978] Improve smoke tests

### PR DESCRIPTION
**Pull request by nickromneydfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-1978

## Changes in this PR:

- Smoke test spec - moved to a single file. Use an environment variable for which domain to be smoke tested, which defaults to Production
- Merge two smoke test workflows into a single workflow with multiple triggers
- Removed duplicated smoke test failure notification from `deploy` workflow (smoke test workflow notifies on failure)
- Add smoke tests to `deploy_branch` workflow (which deploys to dev on pushes to `dev` branch)
- Add smoke tests to `review` workflow (which waits for checks in the review branch)

## Successful runs

This [workflow](https://github.com/DFE-Digital/teaching-vacancies/runs/1872571240?check_suite_focus=true) "dispatched" (triggered) the smoke test workflow:

```
Run benc-uk/workflow-dispatch@v1.1
  with:
    workflow: Smoke Test
    token: ***
    inputs: {"smoke_test_domain": "teaching-vacancies-review-pr-2834.london.cloudapps.digital", "sha": "ac4705f2dc940540eb6723baf7fb5c9dbf9e01e5"}
    ref: teva-1978-smoke-test-review-apps
  env:
    DOCKERHUB_REPOSITORY: dfedigital/teaching-vacancies
    BRANCH: teva-1978-smoke-test-review-apps
    ENVIRONMENT: review-pr-2834
    TAG: review-pr-2834-ac4705f2dc940540eb6723baf7fb5c9dbf9e01e5-20210210154848
```

The smoke test runs _in the branch itself_ https://github.com/DFE-Digital/teaching-vacancies/runs/1872600807?check_suite_focus=true
`Smoke Test teaching-vacancies-review-pr-2834.london.cloudapps.digital ac4705f2dc940540eb6723baf7fb5c9dbf9e01e5`

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2834.london.cloudapps.digital>

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -123,30 +123,20 @@ jobs:
...
-        inputs: '{"environment": "staging", "sha": "${{ github.sha }}"}'
+        inputs: '{"smoke_test_domain": "staging.teaching-vacancies.service.gov.uk", "sha": "${{ github.sha }}"}'
 
     - name: Wait for smoke test
       id: wait_for_smoke_test
       uses: fountainhead/action-wait-for-check@v1.0.0
       with:
         token: ${{ secrets.PERSONAL_TOKEN }}
-        checkName: Smoke Test ${{ github.sha }} on staging # Job name within workflow
+        checkName: Smoke Test staging.teaching-vacancies.service.gov.uk ${{ github.sha }}
```
Comment (saliceti) [inline .github/workflows/deploy.yml:135]: not `smoke_test_domain` variable?

File: .github/workflows/deploy_branch.yml [hunk 2]
```diff
@@ -105,6 +105,23 @@ jobs:
         timeoutSeconds: 720
         intervalSeconds: 15
 
+    - name: Trigger Smoke Test
+      uses: benc-uk/workflow-dispatch@v1.1
+      with:
+        workflow: Smoke Test
+        token: ${{ secrets.PERSONAL_TOKEN }}
+        inputs: '{"smoke_test_domain": "dev.teaching-vacancies.service.gov.uk", "sha": "${{ github.sha }}"}'
```
Comment (saliceti) [inline .github/workflows/deploy_branch.yml:113]: use `${{ env.BRANCH }}` instead?

Comment (nickromneydfe) [reply .github/workflows/deploy_branch.yml]: Thanks. Have adjusted that in lines 113 and 120

**Review by saliceti — APPROVED** [review]

**Pull request merged**